### PR TITLE
added user_commission and tests

### PR DESCRIPTION
**Pull request by mec07**

### CODE REVIEW

File: tests/test_discount.py [hunk 1]
```diff
@@ -1,9 +1,45 @@
...
+        assert discount.code == 'ADULT'
+        assert discount.description == 'Adult standard'
+        assert discount.price_band_code == 'A/pool'
+        assert discount.seatprice == 160.0
+        assert discount.surcharge == 5.5
+        assert discount.is_offer is True
+        assert discount.non_offer_seatprice == 200
+        assert discount.non_offer_surcharge == 6.5
+        assert discount.availability == 6
+        assert discount.user_commission.__dict__ == user_commission.__dict__
```
Comment (nicwest) [inline tests/test_discount.py]: be explicit about what you are expecting. `__dict__` on the user commission object can overwritten and your tests will still pass even though the code may be broken.

The way I have been doing this is to supply the bare minimum amount of data to verify that `from_api_data` is being called correctly then verifying the type and that data is valid

for example:
https://github.com/ingresso-group/pyticketswitch/blob/master/tests/test_order.py#L86-L88
https://github.com/ingresso-group/pyticketswitch/blob/master/tests/test_order.py#L142-L143

this way you test the functionality, it's clear what you are testing, your test remains terse, you get some penetration, and you don't have to mock/patch anything.

File: pyticketswitch/discount.py [hunk 2]
```diff
@@ -28,6 +29,7 @@ def __init__(self, code, description=None, price_band_code=None,
         self.availability = availability
         self.percentage_saving = percentage_saving
         self.absolute_saving = absolute_saving
+        self.user_commission = user_commission
```
Comment (nicwest) [inline pyticketswitch/discount.py]: you need to add this new attribute to the doc string

File: pyticketswitch/client.py [hunk 3]
```diff
@@ -868,7 +868,7 @@ def get_send_methods(self, performance_id):
 
         return send_methods, meta
 
-    def get_discounts(self, performance_id, ticket_type_code, price_band_code):
+    def get_discounts(self, performance_id, ticket_type_code, price_band_code, user_commission=False):
```
Comment (nicwest) [inline pyticketswitch/client.py]: you need to add this new argument to the doc string.

File: tests/test_discount.py [hunk 4]
```diff
@@ -12,8 +48,7 @@ def test_from_api_data(self):
             'sale_surcharge': 5.5,
             'is_offer': True,
             'non_offer_sale_seatprice': 200,
-            'non_offer_sale_surcharge': 6.5,
-            'absolute_saving': 40,
+            'non_offer_sale_surcharge': 6.5, 'absolute_saving': 40,
```
Comment (nicwest) [inline tests/test_discount.py]: formatting.

Comment (mec07) [inline tests/test_discount.py]: well spotted

File: tests/test_discount.py (hunk 1 above)
Comment (mec07) [inline tests/test_discount.py]: Good suggestion, thanks :)

**Pull request merged**